ext/AvizExt/viz/data_envelopment.jl

```diff
@@ -0,0 +1,133 @@
+```
+
+# Arguments
+- `rs` : ResultSet
+- `DEA_output` : output structure from a DEA analysis, carried out using `data_envelopment_analysis`
+- `axis_opts` : Additional options to pass to adjust Axis attributes
+  See: https://docs.makie.org/v0.19/api/index.html#Axis
+- `fig_opts` : Additional options to pass to adjust Figure creation
+  See: https://docs.makie.org/v0.19/api/index.html#Figure
+- `opts` : Aviz options
+    - `frontier_type`, type of frontier to plot: "CRS","VRS" or "FDH".
+    - `line_color`, color to use for best practice frontier.
+    - `data_color`, color to use for data cloud when plotting efficiency frontier.
+    - `frontier_name`, text to label efficiency frontier.
+    - `data_name`, text to label data cloud used to plot efficiency frontier.
+    - `metrics_x_lab` : Name of metric on x axis.
+    - `metrics_y_lab` : Name of metric on y axis.
+    - `scale_eff_y_lab` : Label for scale efficiency.
+    - `scale_eff_y_lab` : Label for technical efficiency.
+"""
+function ADRIA.viz.data_envelopment_analysis(
+    rs::ResultSet, DEA_output::DEAResult;
+    axis_opts::OPT_TYPE=DEFAULT_OPT_TYPE(), fig_opts::OPT_TYPE=DEFAULT_OPT_TYPE(),
+    opts::OPT_TYPE=DEFAULT_OPT_TYPE()
+)
+    f = Figure(; fig_opts...)
+    g = f[1, 1] = GridLayout()
+
+    ADRIA.viz.data_envelopment_analysis!(
+        g, rs, DEA_output; opts=opts, axis_opts=axis_opts
+    )
+
+    return f
+end
+function ADRIA.viz.data_envelopment_analysis!(g::Union{GridLayout,GridPosition},
+    rs::ResultSet, DEA_output::DEAResult; axis_opts::OPT_TYPE=DEFAULT_OPT_TYPE(),
+    opts::OPT_TYPE=DEFAULT_OPT_TYPE()
+)
+    return ADRIA.viz.data_envelopment_analysis!(
+        g, DEA_output; opts=opts, axis_opts=axis_opts
+    )
+end
+function ADRIA.viz.data_envelopment_analysis!(g::Union{GridLayout,GridPosition},
+    DEA_output::DEAResult; axis_opts::OPT_TYPE=DEFAULT_OPT_TYPE(),
+    opts::OPT_TYPE=DEFAULT_OPT_TYPE())
+    frontier_color = get(opts, :frontier_color, :red)
+    data_color = get(opts, :data_color, :black)
+    frontier_name = get(opts, :frontier_name, "Best practice frontier")
+    data_name = get(opts, :data_name, "Scenario data cloud")
+    scale_eff_y_lab = get(opts, :scale_eff_y_lab, L"$\frac{eff_{vrs}}{eff_{crs}}$")
+    tech_eff_y_lab = get(opts, :tech_eff_y_lab, L"$\frac{1}{eff_{vrs}}$")
+    metrics_x_lab = get(opts, :metrics_x_lab, L"$metric 1$")
+    metrics_y_lab = get(opts, :metrics_y_lab, L"$metric 2$")
+
+    # Determines which returns to scale approach is used to select scenario peers
+    # (most efficient scenarios)
+    frontier_type = get(opts, :frontier_type, :vrs_peers)
+
+    Y = DEA_output.Y # Output values
+
+    # Find points on best practice frontier
+    best_practice_scens = getfield(DEA_output, frontier_type).J
+
+    scale_efficiency = DEA_output.crs_vals ./ DEA_output.vrs_vals
+
+    # Plot efficiency frontier and data cloud
+    ax_a = Axis(g[1, 1]; xlabel=metrics_x_lab, ylabel=metrics_y_lab, axis_opts...)
```

Comment:
I think there's a way of making this more interesting, doing the same thing in less lines and extracting part of this code to another function:

If I understand, the reason why you need to save the result of these two following `scatter!` to variables is to render the Legend. You could also render the Legend in a different way:

```
marker_colors = [frontier_color, data_color]
marker_els = [MarkerElement(;color=m_color) for m_color in marker_colors]
Legend(g[1,2], marker_els, [frontier_name, data_name]; framevisible=false)
```

You can even put these three lines inside a separate function and pass `opts` to it so you don't need to extract these colors and names here. 

And if you do that, you don't need to save those `scatter!` results to variables. **Then**, you can extract these lines that you repeat three times to an external function:

```
    # Plot the scale efficiency (ratio of efficiencies assuming CRS vs. assuming VRS)
    ax_b = Axis(g[2, 1]; title="Scale efficiency", ylabel=scale_eff_y_lab, axis_opts...)
    scatter!(ax_b, scale_efficiency; color=data_color)
    scatter!(
        ax_b,
        best_practice_scens,
        scale_efficiency[best_practice_scens];
        color=frontier_color
    )
```

And, again, you can pass `opts` to this function directly and extract the relevant variables inside it so you don't need to extract them here).

Reply:
Still about the legend: I've done something similar here: https://github.com/open-AIMS/ADRIA.jl/blob/3acd2a72d79fab22c169a2337f65d1bde11f46a8/ext/AvizExt/viz/scenarios.jl#L290

And a small digression, hope you don't mind:

That makes me think we should have some standardized way of doing this the function I'm suggesting to you and the function in this link are essentially the same except that one generates a `Legend` with `LineElement`s and the other generates a `Legend` with `MarkerElement`s - and that's something we could address using multiple dispatch probably.

All that to say: maybe not do what I just said in this PR, but if you just extract this code that generates a legend (the one I suggested in the previous comment) to a function called `render_marker_legend`, for example, in the future we can make this better and have just a `render_legend` that can be used for different types of elements.

Reply:
I'll add `render_marker_legend` to data_envelopment.jl` but I'm guessing this will eventually be moved to 'viz.jl' when multiple dispatch is incorporated, so it can be used by both your `scenario.jl` functions and the `data_envelopment.jl` functions?